api/api-materials-v1/src/main/java/com/thoughtworks/go/apiv1/materials/MaterialsAdminControllerV1.java

```diff
@@ -0,0 +1,117 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.materials;
+
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.GsonTransformer;
+import com.thoughtworks.go.config.exceptions.BadRequestException;
+import com.thoughtworks.go.config.exceptions.HttpException;
+import com.thoughtworks.go.server.materials.MaterialUpdateService;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static spark.Spark.*;
+
+@Component
+public class MaterialsAdminControllerV1 extends ApiController implements SparkSpringController {
+
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private final MaterialUpdateService materialUpdateService;
+
+    @Autowired
+    public MaterialsAdminControllerV1(ApiAuthenticationHelper apiAuthenticationHelper, MaterialUpdateService materialUpdateService) {
+        super(ApiVersion.v1);
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.materialUpdateService = materialUpdateService;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.Materials.NOTIFY.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+            before("", mimeType, this::verifyContentType);
+            before("/*", mimeType, this::verifyContentType);
+            before("", this.mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);
+            before("/*", this.mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);
+            post(Routes.Materials.NOTIFY.SVN, mimeType, this::svnNotify);
+            post(Routes.Materials.NOTIFY.GIT, mimeType, this::gitNotify);
+            post(Routes.Materials.NOTIFY.HG, mimeType, this::hgNotify);
+            post(Routes.Materials.NOTIFY.SCM, mimeType, this::scmNotify);
+            exception(HttpException.class, this::httpException);
+        });
+    }
+
+    public String svnNotify(Request req, Response res) throws IOException {
+        JsonReader jsonReader = GsonTransformer.getInstance().jsonReaderFrom(req.body());
+        String uuid = jsonReader.optString("uuid")
```

Comment:
Any specific reason to use `optString` over `getString`?

Reply:
Changed occurrences to `getString`